### DISCUSSION TITLE
Autism and Radio Communication

### DISCUSSION THREAD

**Discussion by BrysonZimmerman** [FakeFishGames/Barotrauma#13651] [Ideas and suggestions]
### The Problem
Autistic people commonly have a limited capacity for experiencing distinct, audible words; once this limit is reached, resulting effects vary from irritability to losing the ability to speak to becoming so overloaded that they are forced to physically leave. This disability often prevents "tuning out" audible words, so that commonly used suggestion is not an option.

Unfortunately, the design of Barotrauma's radio system creates a significant amount of overhead for managing groups of people. From a design perspective, this is an interesting mechanic. From an accessibility perspective, this is an additional barrier that pushes out people with disabilities like autism. Communication is critical and sitting in an unused voice channel, or removing radio comms entirely, adds unnecessary difficulty for a person who would be able to verbally participate when things get spicy but can't handle listening to people chat the whole time.

### Solutions
I've come up with a couple concepts that could improve accessibility while also enhancing gameplay.

My favorite option is a communication override function. Given access control settings to prevent abuse, a hotkey or GUI element could allow a person to cut in to all channels to communicate critical information. This creates room for a player to take on the role of coordinator without having to toggle between several channels. The ability to do this would need to be controlled by the server owner to mitigate trolling. The greatest potential issue I see is the need to bypass the existing convention of one wifi channel at a time, although the tweaks to VoipClient.cs and WifiComponent.cs are minor. The most significant barrier to implementation that I had was building a hotkey setting into the settings menu and figuring out where to watch for the KeyDown event and subsequently set the character's radio headset's WifiComponent's new 'enableMulticast' boolean appropriately. This is a result of me not having access to design documentation showing the way the game's objects interact, combined with a lack of time to build my own documentation.

I see the existence of a dedicated override button, with access control, as a way to enable larger groups to break out into different radio channels but still have a chain of command that is able to cut in across all bands to keep the group functioning without having to all sit in channel 0.

Another option is to integrate with the Mumble API, which would allow Barotrauma to manage individual players' local mute states and positional audio, preserving gameplay elements. The benefit is that a Mumble plugin is entirely opt-in and that it gives players the ability to run their own comms systems while still maintaining. Downsides include integrating with a third party tool, imprecise approximations of how in-game radio feels, punting the hard part to players, and excluding anyone who can't get their group to do the effort of getting Mumble started.

Related, but not equivalent, discussions:
#6147
#5679 

Partial implementation, before getting bogged down figuring out UI and key events
https://github.com/gerbilOFdoom/Barotrauma/commit/27526685725ffb683303456df812851c6619e848

**Comment by GreatBlueMoon** [comment]
this would help with not only the disability but general gameplay.

**Comment by BlackMoons** [comment]
I would also like to add the ability for the server owner (or clients) to turn off the 'headset filter' that greatly degrades audio when using the headset (vs local chat) would be a blessing to those who already have problems understanding what people say.
